### PR TITLE
Fix build issues and more on FreeBSD

### PR DESCRIPTION
**Pull request by alonsobsd**
mixxx was update to 2.3 on FreeBSD but I applied some patches to fix some issues

### CODE REVIEW

**Comment by Be-ing** [comment]
Thanks for this! Can you do an interactive rebase on the 2.3 branch and make the commit messages more descriptive?

Where does the __FREEBSD__ define come from? Our CMake scripts don't define it. Is that always defined when compiling on FreeBSD?

File: src/util/versionstore.cpp [hunk 1]
```diff
@@ -77,6 +77,8 @@ QString VersionStore::platform() {
     QString base = QStringLiteral("Linux");
 #elif defined(__WINDOWS__)
     QString base = QStringLiteral("Windows");
+#elif defined(__FreeBSD__)
+    QString base = QStringLiteral("FreeBSD");
```
Comment (Be-ing) [inline src/util/versionstore.cpp:81]: extra spaces at the end of the line

**Comment by Be-ing** [comment]
Maybe we could use `#ifdef __UNIX__` instead to make this more general?

**Comment by Be-ing** [comment]
Let's use `__BSD__` instead. We already have that set by the build system:
```
elseif(UNIX)
  if(APPLE)
    target_compile_definitions(mixxx-lib PUBLIC __APPLE__)
  else()
    target_compile_definitions(mixxx-lib PUBLIC __UNIX__)
    if(CMAKE_SYSTEM_NAME STREQUAL Linux)
      target_compile_definitions(mixxx-lib PUBLIC __LINUX__)
    elseif(CMAKE_SYSTEM_NAME MATCHES "^.*BSD$")
      target_compile_definitions(mixxx-lib PUBLIC __BSD__)
    endif()
  endif()
endif()
```

File: cmake/modules/FindLibUSB.cmake [hunk 2]
```diff
@@ -57,7 +57,11 @@ find_path(LibUSB_INCLUDE_DIR
 mark_as_advanced(LibUSB_INCLUDE_DIR)
 
 find_library(LibUSB_LIBRARY
+if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+  NAMES usb
+else
   NAMES usb-1.0
+endif()
```
Comment (Be-ing) [inline cmake/modules/FindLibUSB.cmake:64]: I think this should work without any special casing for BSD:
```
  NAMES usb-1.0 usb
```

**Review by uklotzde — CHANGES_REQUESTED** [review]
The proposed changes have to be applied upstream. We are not going to maintain custom patches for 3rd party libs.

**Comment by Be-ing** [comment]
@uklotzde I do not understand your comment. This does not change third party dependencies.

**Comment by Be-ing** [comment]
Oh do you mean the hack for SPSCQue? IIRC upstream already decided they didn't want that so we decided to keep it in Mixxx.

**Comment by uklotzde** [comment]
I have noticed changes in lib for SPSC and Kaitai Struct. If you need to apply patches for FreeBSD you have to do this during packaging. Otherwise please submit the changes upstream.

The PR is not reviewable until it is properly rebased.

**Comment by uklotzde** [comment]
If it only modifies a custom patch of us it is acceptable. I didn't verify this.

**Comment by uklotzde** [comment]
Disclaimer: I am not able to properly review PRs during the next days. I'll try to check occasionally for updates.

**Comment by Be-ing** [comment]
The change to Kaitai Struct should be sent upstream.

**Comment by Be-ing** [comment]
I would prefer the patch for Kaitai Struct to stay in the FreeBSD port until that's merged upstream.

**Comment by Be-ing** [comment]
I redid these changes except the one for the Kaitai Struct file on the 2.3 branch in #4122, #4123, and #4124. Please submit the fix for Kaitai Struct upstream. In the meantime you can keep using that patch in the FreeBSD port. Thanks for bringing these issue to our attention.

**Pull request closed without merging**